LEGO1/mxbitmap.cpp

```diff
@@ -0,0 +1,78 @@
+#include "mxbitmap.h"
+#include "mxbool.h"
+#include "mxpalette.h"
+#include "mxresult.h"
+
+// OFFSET: LEGO1 0x100bc980
+MxBitmap::MxBitmap()
+{
+  this->m_info = NULL;
+  this->m_bmiHeader = NULL;
+  this->m_paletteData = NULL;
+  this->m_data = NULL;
+  this->m_unk18 = 0;
+  this->m_palette = NULL;
+}
+
+// OFFSET: LEGO1 0x100bca10
+MxBitmap::~MxBitmap()
+{
+  if (this->m_info != NULL) {
+    free(this->m_info);
+  }
+  if (this->m_data != NULL) {
+     free(this->m_data);
+  }
+  if (this->m_palette != NULL) {
+     // FIXME: Once m_palette/base is understood
+  }
+  delete m_palette;
+}
+
+// OFFSET: LEGO1 0x100bd1c0
+MxPalette *MxBitmap::CreatePalette()
+{
+  // FIXME: This function needs MxPalette to be completed. Also INFERRING usage of MxBool
+  MxPalette *m_pal = NULL;
+  MxPalette *m_ppal;
+  MxBool m_success = MX_FALSE;
+
+  if(this->m_unk18 == '\0') {
+    // ppal = MxPalette::FromBitmapPalette(this->m_paletteData);
```

Comment:
```suggestion
  if(!this->m_unk18) {
    // ppal = new MxPalette(this->m_paletteData);
```

Reply:
the call says ::FromBitmapPalette, I can check later

Reply:
Oh. It's another call further down that says "if(ppal != NULL)" THEN palette from bitmap palette. You're right, when I get home and have access to a x86 machine again I'll fix this.